src/python/TaskWorker/MasterWorker.py

```diff
@@ -392,6 +392,34 @@ def _lockWork(self, limit, getstatus, setstatus):
 
         return True
 
+    def runCanary(self):
+        # Decide whether to use canary_name based on the canary_load probability.
+        use_canary = random.random() < self.config.TaskWorker.canary_load
```

Comment:
remember to add a comment here and in in the confi. template indicating that canary_fraction is a float in [0.-1.], unless you prefer to change to using a percentage

Reply:
Done